GameEngine/Services/AnimalActions/AnimalActions.cs

```diff
@@ -0,0 +1,325 @@
+        /// <param name="animals">Animals.</param>
+        /// <param name="board">Board.</param>
+        public void AddAnimal(IAnimal newAnimal, List<IAnimal> animals, Board board)
+        {
+            List<NewAnimalCoordinates> freeCells = CalculateFreeCellsToAddAnimal(animals, board);
+            if (board.GameBoard.Length > board.GameBoard.Length * ConstantsRepository.HalfOfBoard)
+            {
+                AddAnimalToList(newAnimal, freeCells, animals);
+            }
+            else
+            {
+                return;
+            }
+        }
+
+        /// <summary>
+        /// Adds animal to list with the new coordinates.
+        /// </summary>
+        /// <param name="newAnimal">New animal.</param>
+        /// <param name="newXCoordinate">New X coordinate.</param>
+        /// <param name="newYCoordinate">New Y coordinate.</param>
+        /// <param name="animals">Animals.</param>
+        private void AddAnimalToList(IAnimal newAnimal,List<NewAnimalCoordinates> freeCells, List<IAnimal> animals)
+        {
+            NewAnimalCoordinates freeCoordinates = freeCells[GenerateRandomCoordinates(0, freeCells.Count)];
+            newAnimal.CoordinateX = freeCoordinates.NewXCoordinate;
+            newAnimal.CoordinateY = freeCoordinates.NewYCoordinate;
+            animals.Add(newAnimal);
+        }
+
+        /// <summary>
+        /// Calculates the free cells on board to add a new animal. 
+        /// </summary>
+        /// <param name="animals">Animals.</param>
+        /// <param name="board">Board.</param>
+        /// <returns>Free cells.</returns>
+        private List<NewAnimalCoordinates> CalculateFreeCellsToAddAnimal(List<IAnimal> animals, Board board)
+        {
+            List<NewAnimalCoordinates> freeCellsToAdd = new List<NewAnimalCoordinates>();
+            NewAnimalCoordinates newCoordinates = new NewAnimalCoordinates();
+
+            for (int newXCoordinate = 0; newXCoordinate < board.GameBoard.GetLength(0); newXCoordinate++)
+            {
+                for (int newYCoordinate = 0; newYCoordinate < board.GameBoard.GetLength(1); newYCoordinate++)
+                {
+                    if (!board.IsOffBoard(newXCoordinate, newYCoordinate, board.GameBoard) &&
+                        !IsNotFreePlace(newXCoordinate, newYCoordinate, animals))
+                    {
+                        newCoordinates = new NewAnimalCoordinates();
+                        newCoordinates.NewXCoordinate = newXCoordinate;
+                        newCoordinates.NewYCoordinate = newYCoordinate;
+                        freeCellsToAdd.Add(newCoordinates);
+                    }
+                }
+            }
+
+            return freeCellsToAdd;
+        }
+
+        /// <summary>
+        /// Moves animals on the board.
+        /// </summary>
+        /// <param name="board">Board.</param>
+        /// <param name="animals">Animals.</param>
+        public void Move(Board board, List<IAnimal> animals)
```

Comment:
Here you have method Move for both antelope and lion.
If you will have to add a new animal this method will get rather big.
What you can do is create a new abstract class Animal and add there all properties that animals have in common.
In your case 

```
        public int Id { get; set; }
        public int CoordinateX { get; set; }
        public int CoordinateY { get; set; }
        public AnimalType Type { get; set; }
        public int Vision { get; set; }
        public bool IsDead { get; set; }
        public string Letter { get; set; }

        private static int id = 1;
        private int GenerateId()
        {
            return id++;
        }
```
these properties are present both for antelope and lion so make abstract animal class so lion and antelope should inherit from a animal class. Also, you can put methods that are common for all animals there. And override methods that differ for the corresponding animal.